### PR TITLE
`Development`: Remove unused code

### PR DESCRIPTION
**Pull request by reschandreas**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.
### Motivation and Context

I took a look at our [unused code findings](https://teamscale.io/findings.html#list/GitHub-ls1intum-Artemis/?t=develop%3AHEAD&filter=Comprehensibility%2FUnused+Code&assessment-filters-empty=true&included-paths-empty=true&excluded-paths-empty=true&guideline-rules-empty=true&invert=true) and decided to fix all of them that I can safely remove without changing the templates or functionality of Artemis.
### Description

I removed several variables and autowired code that is never used and could potentially break in the future as it adds to the code complexity.

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

## Summary by CodeRabbit

- **Refactor**
  - Simplified service constructors by removing unnecessary parameters.
  - Streamlined quiz exercise import process by eliminating file handling parameters.
  - Enhanced performance and scalability of migration operations.
  - Updated permission management methods to improve clarity and maintainability.

- **Bug Fixes**
  - Adjusted exercise import and result saving methods to ensure correct behavior.

- **Chores**
  - Cleaned up unused imports and fields across various tests and services.
  - Removed unnecessary logging setup from multiple service classes.

- **Documentation**
  - Added a comment to investigate potential configuration issues.

- **Style**
  - Standardized method signatures and refactored code for better readability.

### CODE REVIEW

**Review by Strohgelaender — DISMISSED** [review]
lgtm

File: src/test/java/de/tum/in/www1/artemis/post/ConversationUtilService.java [hunk 1]
```diff
@@ -625,7 +625,7 @@ public Conversation addOneMessageForUserInCourse(String login, Course course, St
         channel.setCreator(message.getAuthor());
         addReactionForUserToPost(login, message);
         conversationRepository.save(channel);
-        message = postRepository.save(message);
+        postRepository.save(message);
```
[the remark this answers is not in the document]
Comment (reschandreas) [reply src/test/java/de/tum/in/www1/artemis/post/ConversationUtilService.java:628]: is that part of this PR?

File: src/main/java/de/tum/in/www1/artemis/web/rest/QuizExerciseResource.java [hunk 2]
```diff
@@ -676,7 +676,7 @@ public ResponseEntity<SearchResultPageDTO<QuizExercise>> getAllExercisesOnPage(P
     @PostMapping(value = "quiz-exercises/import/{sourceExerciseId}", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     @EnforceAtLeastEditor
     public ResponseEntity<QuizExercise> importExercise(@PathVariable long sourceExerciseId, @RequestPart("exercise") QuizExercise importedExercise,
-            @RequestPart(value = "files", required = false) List<MultipartFile> files) throws URISyntaxException, IOException {
+            @RequestPart(value = "files", required = false) List<MultipartFile> files) throws URISyntaxException {
```
[the remark this answers is not in the document]
Comment (reschandreas) [reply src/main/java/de/tum/in/www1/artemis/web/rest/QuizExerciseResource.java:679]: the files parameter is still there

**Review by milljoniaer — DISMISSED** [review]
LGTM thanks for removing these 🚀

**Review by milljoniaer — APPROVED** [review]
reapprove after code changes

**Review by jakubriegel — COMMENTED** [review]
Code looks (and compiles) good 👍🏻 

Teamscale complains about 3 tests. Most likely this is not connected to this PR, but it will be better to check it before merging

**Comment by reschandreas** [comment]
@jakubriegel, what three tests do you mean?

**Comment by jakubriegel** [comment]
> @jakubriegel, what three tests do you mean?

You can check them in Teamscale report: https://teamscale.io/activity.html#merge-requests/GitHub-ls1intum-Artemis/ls1intum%2FArtemis%2F7864

**Comment by reschandreas** [comment]
@jakubriegel those are findings not really introduced by this PR and I would argue out of scope for this PR as I only want to remove unused code

**Comment by jakubriegel** [comment]
> @jakubriegel those are findings not really introduced by this PR and I would argue out of scope for this PR as I only want to remove unused code

That's ok. Then you can ignore ignore them on Teamscale

**Review by jakubriegel — APPROVED** [review]

**Review by laurenzfb — APPROVED** [review]
lgtm

**Review by rstief — APPROVED** [review]
Code looks good 👍

**Review by milljoniaer — APPROVED** [review]
reapprove after code changes

**Pull request merged**